spec/ndx-patterned-ogen.extensions.yaml

```
@@ -1,8 +1,165 @@
+    attributes:
+      - name: description
+        dtype: text
+        doc: Description of the scanning or scanless method for shaping optogenetic light. Examples include diffraction limited points, 3D shot, disks, etc.
+        required: false
+      - name: sweep_size
+        dtype: numeric
+        doc:
+          Size of the scanning sweep pattern in micrometers. If a scalar is provided, the sweep pattern is
+          assumed to be a circle (for 2D patterns) or cylinder (for 3D patterns), with diameter 'sweep_size'.
+          If 'sweep_size' is a two or three dimensional array, the the sweep pattern is assumed to be a
+          rectangle or cuboid, with dimensions [width, height] or [width, height, depth].
+        required: false
+      - name: sweep_mask
+        dtype: numeric
+        doc:
+          Scanning sweep pattern designated using a mask of size [width, height] (2D stimulation) or [width,
+          height, depth] (3D stimulation), where for a given pixel a value of 1 indicates stimulation, and a
+          value of 0 indicates no stimulation.
+        required: false
+  - neurodata_type_def: SpiralScanning
+    neurodata_type_inc: LabMetaData
+    doc: Container to store the parameters defining a spiral scanning pattern.
+    attributes:
+      - name: description
+        dtype: text
+        doc: Description of the pattern.
+        required: false
+      - name: diameter
+        dtype: numeric
+        doc: Spiral diameter of each sweep (in micrometers).
+      - name: height
+        dtype: numeric
+        doc: Spiral height of each sweep (in micrometers).
+        required: false
+      - name: number_of_revolutions
+        dtype: numeric
+        doc: Number of turns within a spiral.
+  - neurodata_type_def: TemporalFocusing
+    neurodata_type_inc: LabMetaData
+    doc: Container to store the parameters defining a temporal focusing beam-shaping.
+    attributes:
+      - name: description
+        dtype: text
+        doc: Description of the pattern.
+        required: false
+      - name: lateral_point_spread_function
+        dtype: text
+        doc:
+          Estimated lateral spatial profile or point spread function, expressed as
+          mean [um] ± s.d [um].
+      - name: axial_point_spread_function
+        dtype: text
+        doc:
+          Estimated axial spatial profile or point spread function, expressed as mean
+          [um]± s.d [um].
+  - neurodata_type_def: PatternedOptogeneticStimulusSite
+    neurodata_type_inc: OptogeneticStimulusSite
+    doc: Patterned optogenetic stimulus site.
+    attributes:
+      - name: effector
+        dtype: text
+        doc: Light-activated effector protein expressed by the targeted cell (e.g., ChR2).
+    links:
+      - name: spatial_light_modulator
+        target_type: SpatialLightModulator
+        doc: SpatialLightModulator used to generate holographic pattern.
```

```suggestion
        doc: Spatial light modulator used to generate holographic pattern.
```

Adjusting verbage to be more human-readable as opposed to code/schema references

---

If you agree, there's a lot more such references throughout the spec beyond the suggestions I've made so far

---

Sure, I willl adjust the others. Thanks